src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -82,6 +88,7 @@ private static ExecutorService getSharedExecutor() {
   private final Executor executor;
   private final CompletableFuture<Void> closeFuture;
   private final AtomicInteger chunkedBytesSent = new AtomicInteger(0);
+  private final AtomicBoolean requiresRset = new AtomicBoolean(false);
```

Comment:
if you don't need atomic primitives, I would just use `volatile boolean`. Saves a few bytes and limits what your intended uses are